repository/src/lib.rs

```diff
@@ -156,13 +156,184 @@ impl<T: RawRepository> DistributedRepository<T> {
+                })?;
+
+            // Verify all the incoming commits and apply them to the local repository only if they are valid.
+            let mut verifier =
+                CommitSequenceVerifier::new(last_header.clone(), reserved_state.clone())
+                    .map_err(|e| anyhow!("failed to create a verifier: {}", e))?;
+            for (new_commit, _) in &new_commits {
+                if verifier.apply_commit(new_commit).is_err() {
+                    continue 'branch;
+                }
+            }
+            let branch_tip_commit = &new_commits.last().unwrap().0;
+            if let Commit::Agenda(_) = branch_tip_commit {
+                // If branch tip commit is Agenda commit, create a new agenda branch
+                let branch_tip_commit_hash = self.raw.locate_branch(branch).await?;
+                let branch_name = "a-"; //TODO : change # into number
+                let branch_name = format!("{}{}", branch_name, branch_tip_commit_hash);
+                self.raw
+                    .create_branch(branch_name, branch_tip_commit_hash)
+                    .await?;
+            } else if let Commit::AgendaProof(_) = branch_tip_commit {
+                // Else if branch tip commit is AgendaProof commit, create a new agenda branch
+                let branch_tip_commit_hash = self.raw.locate_branch(branch).await?;
+                let branch_name = "a-#"; //TODO : change # into number
+                let branch_name = format!("{}{}", branch_name, branch_tip_commit_hash);
+                self.raw
+                    .create_branch(branch_name, branch_tip_commit_hash)
+                    .await?;
+            } else if let Commit::Block(block_header) = branch_tip_commit {
+                // find fp, if there is fp then move finalized branch
+                let fp_commit_hash = self.raw.locate_branch(FP_BRANCH_NAME.into()).await?;
+                let fp_semantic_commit = self.raw.read_semantic_commit(fp_commit_hash).await?;
+                let finalization_proof: FinalizationProof =
+                    serde_json::from_str(&fp_semantic_commit.body).unwrap();
+                let result = verify::verify_finalization_proof(
+                    block_header_vec.get(0).unwrap(),
+                    &finalization_proof,
+                );
+                //if the finalization proof is right, we push it to a vector to check
+                //if we can't find right finalization proof, we create a new b# branch
+                match result {
+                    Ok(()) => {
+                        block_header_vec.push(block_header.clone());
+                        finalized_branches.push(branch);
+                    }
+                    Err(_) => {
+                        let branch_tip_commit_hash = self.raw.locate_branch(branch).await?;
+                        let branch_name = "b-#";
+                        let branch_name = format!("{}{}", branch_name, branch_tip_commit_hash);
```

Comment:
ditto except hash value of the block commit

Reply:
fixed.